File: Domotización - Interfaces/Domotización - Interfaces/Ventana.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domotización___Interfaces
{
    internal abstract class Ventana
    {
        private int numVentana;
        private bool bloqueada = false;

        public int NumVentana { get { return numVentana; } set { numVentana = value; } }

        public bool Bloqueada { get { return bloqueada; } set { bloqueada = value; } }

        protected Ventana(int numVentana)
        {
            this.numVentana = numVentana;
            bloqueada = false;
        }

        public override string ToString()
        {
            return $"Número Ventana: {NumVentana} Bloqueada: {(Bloqueada ? "Bloqueada" : "Desbloqueada")}";
        }

        public abstract void bloquear();

        public abstract void desbloquear();

    }
}

File: DomotizacionInterfaces/src/main/java/org/example/Ventana.java
package org.example;

public class Ventana {

    protected Ventana(int numVentana)
    {
        this.numVentana = numVentana;
        bloqueada = false;
    }

    private int numVentana;
    private boolean bloqueada = false;

    public int getNumVentana() {
        return numVentana;
    }

    public void setNumVentana(int numVentana) {
        this.numVentana = numVentana;
    }

    public boolean isBloqueada() {
        return bloqueada;
    }

    public void setBloqueada(boolean bloqueada) {
        this.bloqueada = bloqueada;
    }

    @Override
    public String toString() {
        return "Número Ventana: " + numVentana + " || Bloqueada: " + (isBloqueada() ? "Bloqueada" : "Desbloqueada");
    }

    public void bloquear() {}

    public void desbloquear() {}

}
